### PR TITLE
Slightly simplify some BlockParser logic

### PR DESCRIPTION
**Pull request by chirhonul**
Also add TODO about concern in validating block header vs just
block height.

### CODE REVIEW

**Comment by ripcurlx** [comment]
utACK

File: src/main/java/bisq/core/dao/node/parser/BlockParser.java [hunk 1]
```diff
@@ -134,8 +134,10 @@ private void validateIfBlockIsConnecting(RawBlock rawBlock) throws BlockNotConne
         }
     }
 
-    private boolean isBlockNotAlreadyAdded(RawBlock rawBlock) {
-        return !bsqStateService.getBlockAtHeight(rawBlock.getHeight()).isPresent();
+    private boolean isBlockAlreadyAdded(RawBlock rawBlock) {
+        // TODO(chirhonul): shouldn't we verify that we know about the blockHash, not just that the
+        // block heights are the same? how do we handle chainsplits otherwise?
```
Comment (ManfredKarrer) [inline src/main/java/bisq/core/dao/node/parser/BlockParser.java]: Yes, true.

Comment (chirhonul) [reply src/main/java/bisq/core/dao/node/parser/BlockParser.java]: That seems beyond the scope of the current PR (which is a refactor with no functional changes), which is why I added it as a `TODO` here. I'd be happy to address it in a follow-up PR.

**Comment by chirhonul** [comment]
Thanks for the review!

@ManfredKarrer: I'd prefer to change the functionality of `isBlockAlreadyAdded()` to make it more robust in a follow-up PR (see comment inline). Please take another look and see if you agree that we can merge this (once current release is out) as-is.

**Comment by chirhonul** [comment]
I've made the changes mentioned as a `TODO` in a branch based on this changeset:

* https://github.com/chirhonul/bisq-core/compare/1808-block-parse-tweaks...chirhonul:1808-check-blockhash?expand=1

I will open the second PR when this one has been merged.

**Comment by ManfredKarrer** [comment]
utACK

File: src/main/java/bisq/core/dao/node/parser/BlockParser.java (hunk 1 above)
Comment (ManfredKarrer) [reply src/main/java/bisq/core/dao/node/parser/BlockParser.java]: Ok.

**Pull request merged**